lib/chutney/TorNet.py

```diff
@@ -1300,7 +1327,12 @@ def getNodeDirInfoStatusPattern(self, dir_format):
            the ed25519 key. (Or RSA block matching, which is hard.)
         """
         nickname = self.getNick()
-
+        ed25519_id = None
+        # old tor versions do not have an ed25519-id
+        if 'ed25519_id' in self._env:
```

Comment:
_env doesn't work like a dict, so you'll need to catch the KeyError yourself, and return None.

Try writing a function like getBridge(), but call it getEd25519Id():
https://github.com/torproject/chutney/blob/76981f086d29176c05d060acc8acb9b4d01886cd/lib/chutney/TorNet.py#L859